core/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java

```diff
@@ -277,66 +277,6 @@ public void testTwoNodesSingleDoc() throws Exception {
         }
     }
 
-    public void testDanglingIndicesConflictWithAlias() throws Exception {
-        logger.info("--> starting two nodes");
-        internalCluster().startNodesAsync(2).get();
-
-        logger.info("--> indexing a simple document");
-        client().prepareIndex("test", "type1", "1").setSource("field1", "value1").setRefresh(true).execute().actionGet();
-
-        logger.info("--> waiting for green status");
-        ensureGreen();
-
-        logger.info("--> verify 1 doc in the index");
-        for (int i = 0; i < 10; i++) {
-            assertHitCount(client().prepareSearch().setQuery(matchAllQuery()).get(), 1L);
-        }
-        assertThat(client().prepareGet("test", "type1", "1").execute().actionGet().isExists(), equalTo(true));
-
-        internalCluster().stopRandomNonMasterNode();
-
-        // wait for master to processed node left (so delete won't timeout waiting for it)
-        assertFalse(client().admin().cluster().prepareHealth().setWaitForNodes("1").get().isTimedOut());
-
-        logger.info("--> deleting index");
-        assertAcked(client().admin().indices().prepareDelete("test"));
-
-        index("test2", "type1", "2", "{}");
-
-        logger.info("--> creating index with an alias");
-        assertAcked(client().admin().indices().prepareAliases().addAlias("test2", "test"));
-
-        logger.info("--> starting node back up");
-        internalCluster().startNode();
-
-        ensureGreen();
-
-        // make sure that any other events were processed
-        assertFalse(client().admin().cluster().prepareHealth().setWaitForRelocatingShards(0).setWaitForEvents(Priority.LANGUID).get()
-            .isTimedOut());
-
-        logger.info("--> verify we read the right thing through alias");
-        assertThat(client().prepareGet("test", "type1", "2").execute().actionGet().isExists(), equalTo(true));
-
-        logger.info("--> deleting alias");
-        assertAcked(client().admin().indices().prepareAliases().removeAlias("test2", "test"));
-
-        logger.info("--> waiting for dangling index to be imported");
-
-        assertBusy(new Runnable() {
-            @Override
-            public void run() {
-                assertTrue(client().admin().indices().prepareExists("test").execute().actionGet().isExists());
-            }
-        });
-
-        ensureGreen();
-
-        logger.info("--> verifying dangling index contains doc");
-
-        assertThat(client().prepareGet("test", "type1", "1").execute().actionGet().isExists(), equalTo(true));
-    }
-
```

Comment:
@bleskes @s1monw I removed this test as it was failing for this PR, but I wanted to validate the logic with you before making a decision on how to proceed.  In the new scenario of explicit index tombstones in the cluster state, I do not see why a dangling index should be imported in this case (hence, the reason I deleted the test for the time being).  In this test, first an index named `test` is created and a document is indexed into it.  This index will have some UUID, lets call it `X`.  Then the non master node goes offline, and `test` with UUID `X` is deleted.  Then, an index named `test2` is created with a UUID, lets call it `Y`, and this new index is assigned an alias of `test`.  Then, the non master node is restarted.  At this point, when it is restarted, it receives cluster state from master with a tombstone for index `test` of UUID `X`.  So the non master node sees that it has this index metadata on disk and deletes it.  Therefore, there is nothing to import at the dangling index stage, even when the alias for `test2` is dropped.  I'm not sure why we would want that index to be imported as `test` in the first place, as it was explicitly deleted and only `test2` should exist.

That was my logic when thinking that this test is no longer valid, but I could well be wrong and am not sure, so would appreciate any validation either way.